src/commands/deploy.rs

```diff
@@ -125,10 +125,9 @@ impl DeployCommand {
         };
 
         let mut login_connection: LoginConnection = serde_json::from_str(&data)?;
-
-        let expiration_date = DateTime::parse_from_rfc3339(&login_connection.expiration)?;
-        let now: DateTime<Utc> = Utc::now();
-        if now > expiration_date {
+        let expiration = login_connection.expiration.clone();
+        if expiration.is_some() && Utc::now() > DateTime::parse_from_rfc3339(&expiration.unwrap())?
```

Comment:
Suggestion: move this logic out to a `has_expired(&LoginConnection)` function - the nesting of brackets and connectives is otherwise fiddly for a reader to parse.

Reply:
sure, done